scale out pd cause cdc panic

### What did you do?

scale out 1 pd instance

### What did you expect to see?

cdc works normal

### What did you see instead?

```
panic: runtime error: index out of range [0] with length 0

goroutine 488 [running]:
github.com/pingcap/tiflow/pkg/pdutil.(*pdAPIClient).CollectMemberEndpoints(0xc00110bd58?, {0x3d72bc8?, 0xc002240940?})
	github.com/pingcap/tiflow/pkg/pdutil/api_client.go:211 +0x169
github.com/pingcap/tiflow/cdc/server.(*server).etcdHealthChecker(0xc000b9d4a0, {0x3d72bc8?, 0xc002240940})
	github.com/pingcap/tiflow/cdc/server/server.go:250 +0x39c
github.com/pingcap/tiflow/cdc/server.(*server).run.func2()
	github.com/pingcap/tiflow/cdc/server/server.go:281 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:72 +0xa5
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
Release Version: v6.3.0-master
Git Commit Hash: 5fa97cdc28058573f32c89ac138eda2d866d8abe
Git Branch: scheduler-burst-add-by-batch
UTC Build Time: 2022-08-24 08:12:58
Go Version: go version go1.18 linux/amd64
Failpoint Build: false
```

--- a/pkg/pdutil/api_client.go
+++ b/pkg/pdutil/api_client.go
@@ -208,7 +208,10 @@ func (pc *pdAPIClient) CollectMemberEndpoints(ctx context.Context) ([]string, er
 	}
 	result := make([]string, 0, len(members))
 	for _, m := range members {
-		result = append(result, m.GetClientUrls()[0])
+		clientUrls := m.GetClientUrls()
+		if len(clientUrls) > 0 {
+			result = append(result, clientUrls[0])
+		}
 	}
 	return result, nil
 }

